### DISCUSSION TITLE
beforeUnload Not Working with Socket.io

### DISCUSSION THREAD

**Discussion by Shivam-Shandilya1** [socketio/socket.io#5053] [Q&A]
Before closing the tab or window I ask the user either they want to leave? If they confirms The socket should disconnect otherwise there will be no change.

I am using
"socket.io": "^4.7.5",
"socket.io-client": "^4.7.5",
 "socket.io-redis": "^6.1.1",
 
 and Browser: Chrome
 
 But actual result:  after closing tab, the socket immediately got disconnected even without listening to the user confirmation.

**Comment by darrachequesne** [comment]
Hi! The `closeOnBeforeunload` option controls whether the socket is disconnected when receiving a "beforeunload" event. It now defaults to `false` since version 4.7.1, which means that the socket won't be automatically disconnected.

Reference: https://socket.io/docs/v4/client-options/#closeonbeforeunload
